File: Tarefa3/main.py
from utils import integracao


if __name__ == "__main__":

    print("---------------------------------------------")
    print("-------------------Tarefa 3------------------")
    print("---------------------------------------------\n")

    programa = True

    while programa:
        print("A funcao usada sera: (sen(2x) + 4*(x)^2 + 3x)^2\n")
        xi = int(input("Escreva o valor de xi:"))
        xf = int(input("Escreva o valor de xf:"))
        epilson = float(input("Escreva o valor do erro:"))

        print("Escolha a abordagem desejada: ")
        print("1: Abordagem Aberta ")
        print("2: Abordagem Fechada ")
        escolha = int(input("Digite o número da abordagem que você quer: "))

        print("Escolha o grau desejado: ")
        print("1: 1°grau ")
        print("2: 2°grau ")
        print("3: 3°grau ")
        print("4: 4°grau ")
        grau = int(input("Digite grau que você quer: "))
        
        if escolha == 1:
            if grau == 1:
                print("Você escolheu a Abordagem Aberta de 1°grau")
                integracao(xi, xf, epilson, escolha, grau)
            if grau == 2:
                print("Você escolheu a Abordagem Aberta de 2°grau")
                integracao(xi, xf, epilson, escolha, grau)
            if grau == 3:
                print("Você escolheu a Abordagem Aberta de 3°grau")
                integracao(xi, xf, epilson, escolha, grau)
            if grau == 4:
                print("Você escolheu a Abordagem Aberta de 4°grau")
                integracao(xi, xf, epilson, escolha, grau)

        if escolha == 2:
            if grau == 1:
                print("Você escolheu a Abordagem Fechado de 1°grau")
                integracao(xi, xf, epilson, escolha, grau)
            if grau == 2:
                print("Você escolheu a Abordagem Fechado de 2°grau")
                integracao(xi, xf, epilson, escolha, grau)
            if grau == 3:
                print("Você escolheu a Abordagem Fechado de 3°grau")
                integracao(xi, xf, epilson, escolha, grau)
            if grau == 4:
                print("Você escolheu a Abordagem Fechado de 4°grau")
                integracao(xi, xf, epilson, escolha, grau)

        print("\n---------------------------------------------")
        print("1-Rodar o programa novamente ")
        print("2-Sair do programa")
        print("---------------------------------------------")

        novamente = int(input("Digite a sua escolha?"))

        if novamente == 1:
            programa = True
        if novamente == 2:
            programa = False

File: Tarefa3/main.cpp
#include <iostream>
#include <cmath>
#include "utils.cpp"

using namespace std;

main(){
    cout << "---------------------------------------------" << endl;
    cout << "-------------------Tarefa 3------------------" << endl;
    cout << "---------------------------------------------" << endl;

    bool programa = true;

    while(programa){
        cout << ("A funcao usada sera: (sen(2x) + 4*(x)^2 + 3x)^2\n");
        int xi;
        cout << "Escreva o valor de xi:";
        cin >> xi;
        int xf;
        cout << "Escreva o valor de xf:";
        cin >> xf;
        double epilson;
        cout << "Escreva o valor do erro:";
        cin >> epilson;
        cout << "Escolha a abordagem desejada:\n";
        cout << "1: Abordagem Aberta" << endl;
        cout << "2: Abordagem Fechada" << endl;
        int escolha;
        cout << "Digite o número da abordagem que voce quer escolher" << endl;
        cin >> escolha;

        cout << "Escolha o grau desejado::\n";
        cout << "1: 1 grau" << endl;
        cout << "2: 2 grau" << endl;
        cout << "3: 3 grau" << endl;
        cout << "4: 4 grau" << endl;
        int grau;
        cout << "Digite o número da abordagem que voce quer escolher" << endl;
        cin >> grau;

        if(escolha == 1){
            if(grau == 1){
                cout << "Voce escolheu a Abordagem Aberta de 1°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else if(grau ==2){
                cout << "Voce escolheu a Abordagem Aberta de 2°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else if(grau ==3){
                cout << "Voce escolheu a Abordagem Aberta de 3°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else if(grau ==4){
                cout << "Voce escolheu a Abordagem Aberta de 4°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else{
                cout << "Essa opção não existe" << endl;
                break;
            }
        }

        if(escolha == 2){
            if(grau == 1){
                cout << "Voce escolheu a Abordagem Fechada de 1°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else if(grau ==2){    
                cout << "Voce escolheu a Abordagem Fechada de 2°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else if(grau ==3){
                cout << "Voce escolheu a Abordagem Fechada de 3°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else if(grau ==4){
                cout << "Voce escolheu a Abordagem Fechada de 4°grau" << endl;
                integracao(xi, xf, epilson, escolha, grau);
            } else{
                cout << "Essa opção não existe" << endl;
                break;
            }
        }
        cout << "Quer continuar com o programa?\n";
        cout << "1: Sim" << endl;
        cout << "2: Não" << endl;
        int resposta;
        cout << "Digite o número da resposta:" << endl;
        cin >> resposta;
        if(resposta == 2){
            programa=false;
        }
    } 
}